forem/forem
# Extracted the logo uploader into its own partial

## What type of PR is this? (check all applicable)


## Description

This is pre-work to introduce the logo uploader field to the Images sections of the admin -> customization -> config section.

## Related Tickets & Documents

#15618

## QA Instructions, Screenshots, Recordings

Everything should behave as it did previously in the creator settings form.

1. Ensure you're logged out from the Forem instance.
2. Via the rails console, `bin/rails console`, Set up the database to be in a state where you are about to create the Forem creator role

```ruby
# Enable the feature flag
FeatureFlag.enable(:creator_onboarding)

# Delete all users 
User.destroy_all

# Set `waiting_on_first_user` to be true
Settings::General.waiting_on_first_user = true

# If you've logged in before set the mascot id to nil
Settings::General.mascot_user_id = nil

# If you've logged in before set the logo SVG to nil
Settings::General.logo_svg = nil
```

3. Click on the login link. Since the feature flag is enabled, you'll be redirected to Creator Signup Form.
4. Fill in the necessary information and submit the form.
5. On the next page select an image for the logo.
6. Ensure both the Terms and Conditions and Code of Conduct checkboxes are checked off and submit the form.
7. Notice that the logo loads in the header.
8. You can redo this process with multiple image types.

### UI accessibility concerns?

N/A

## Added/updated tests?

- [x] No, and this is why: Existing tests apply to this refactor.

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![A four piece puzzle being put together](https://media.giphy.com/media/RIqh9nbpblgvWvd6ZK/giphy.gif)
